thrift.js

```diff
@@ -82,6 +93,7 @@ function Thrift(options) {
     if (options.allowFilesystemAccess) {
         this.fs = fs;
     }
+    this.loader = (this.fs && this.fs.load) || this._syncFileLoader(this.fs);
```

Comment:
There are a couple smells here. Functions should be named as verbs, objects as nouns. If this were to remain, it should be called `load`.

However `fs` is effectively a "loader" so an alternate framing of the problem would reveal that it would be better to:

```
this.fs = this.fs || this._syncFs(this.fs);
```

Unless the implication is that `fs` is synchronous and `loader` is asynchronous. But, I don’t buy that. Just as well if not better to detect that based on the methods implemented on `fs`.

The other smell, possibly irrelevant, is that we should not depend on `this.fs.load` to be bound to `this.fs`. It happens to be a closure, but we should not take that as a guarantee. A mock filesystem need not have bound methods, for example. I think this makes the stronger case for using `this.fs` directly.